### ISSUE TITLE
Custom Helper not found when {{ moustache }} is split over lines after upgrading to 4.1.1/4.1.2

### ISSUE THREAD

**Issue by v79** [jknack/handlebars.java#671]
I've a fairly simple custom helper class which worked fine in 4.1.0 but is failing to register in 4.1.1 or 4.1.2. I'm writing in Kotlin, which has always worked fine and other helpers seem to be registered.

*tl;dr; a moustache calling a custom helper fails when it is split over multiple lines, like {{ localDate\nthis.date }} in 4.1.1 and 4.1.2; worked in 4.1.0.*

I register my helpers:

```kotlin
val dateFormat = "dd/MMM/yyyy"
hbRenderer = Handlebars(loader)
hbRenderer.registerHelper("forEach", ForEachHelper())
hbRenderer.registerHelper("paginate", Paginate())
hbRenderer.registerHelper("localDate", LocalDateFormatter(dateFormat)) // <-- this is the one which fails
hbRenderer.registerHelper("capitalize",StringHelpers.capitalize)
hbRenderer.registerHelper("upper",StringHelpers.upper)
hbRenderer.registerHelper("slugify", StringHelpers.slugify)
```
When I debug, the `hbRenderer.registry.helpers hashmap` does contain my `LocalDateFormatter` class.

That class is roughly defined like this:
```kotlin
class LocalDateFormatter(val inputFormat: String = "dd/MM/yyyy") : Helper<Any> {
   override fun apply(value: Any?, options: Options): CharSequence {
   val format = options.param(0, options.hash<Any>("format", inputFormat)) as String
		val outputFormatter = when (format) {
			ISO_DATE -> {
				DateTimeFormatter.ISO_DATE
			}
			// more cases here....
		val result = outputFormatter.format(value as LocalDate)
		return result
    }
  }
}
```
The stacktrace error message doesn't say much:

com.github.jknack.handlebars.HandlebarsException: inline@3a8c7071:65:69: could not find helper: 'localDate	datetime="{{localDate this.date "ISO_LOCAL_DATE"}}"> {{ localDate                                                             ^
													this.date'

The calling HTML template looks like this (I've tried to retain the newlines/spacing...):
```html
<span class="tag is-rounded"><time
													datetime="{{localDate this.date "ISO_LOCAL_DATE"}}"> {{ localDate
													this.date "EEE, dd MMM yyyy" }}</time></span>
```

If I strip out all the newlines and extra spacing, it works correctly.

There seems to have been changes in behaviour across 4.1.0, 4.1.1 and 4.1.2.

With **Test 1**, helper call split over lines has a parameter:
```html
<span class="tag is-rounded"><time
													datetime="{{localDate this.date "ISO_LOCAL_DATE"}}"> {{ localDate
													this.date "EEE, dd MMM yyyy" }}</time></span>
```
With **Test 2**, helper call split over lines does not have a parameter:
```html
<time
												datetime="{{localDate newer.date "ISO_LOCAL_DATE"}}">{{localDate
												newer.date}}</time>
```

| Version |  Test 1| Test 2 |
| -------- |:-----:| -----:|
| 4.1.0      | ✔ | ✔ |
| 4.1.1      | ❌|  ✔ |
| 4.1.2      | ❌|  ❌ |

**Comment by elGatoMantocko** [comment]
+1 to this. We just updated to 4.1.2 and noticed similar issues. What architecture are you on? I've only been able to repro on Windows machines. It might be caused by carriage returns.

**Comment by v79** [comment]
> 
> 
> +1 to this. We just updated to 4.1.2 and noticed similar issues. What architecture are you on? I've only been able to repro on Windows machines. It might be caused by carriage returns.

I am using Windows 10, 64bit.

**Comment by jknack** [comment]
Fixed in 4.2.0
